common/utils.py

```diff
@@ -1,19 +1,55 @@
 import binascii
+import enum
 from copy import deepcopy
 from functools import wraps
 
 from django.conf import settings
 from django.core.exceptions import PermissionDenied
 from django.db import transaction
 from graphene import Node
-from graphql.execution.base import ResolveInfo
+from graphql.type import GraphQLResolveInfo
 from graphql_relay import from_global_id, to_global_id
 
 from kukkuu import __version__
 from kukkuu.exceptions import DataValidationError, ObjectDoesNotExistError
 from kukkuu.settings import REVISION
 
 
+def is_enum_value(value):
+    """
+    Check if a value is an enum value, e.g. TestEnum.FI
+    where TestEnum derives from enum.Enum or graphene.Enum.
+    """
+    # Works both for enum.Enum and graphene.Enum
+    return type(type(value)) is enum.EnumMeta
+
+
+def deepfix_enum_values(data):
+    """
+    Fix enum values recursively in/out of dictionaries, lists, sets, and tuples.
+    """
+    if isinstance(data, dict):
+        return {deepfix_enum_values(k): deepfix_enum_values(v) for k, v in data.items()}
+    elif isinstance(data, (list, set, tuple)):
+        return type(data)(deepfix_enum_values(v) for v in data)
+    elif is_enum_value(data):
+        return data.value
+    else:
+        return data
+
+
+def map_enums_to_values_in_kwargs(method):
+    """
+    Decorator that maps enums to their values in keyword arguments.
+    """
+
+    def wrapper(*args, **kwargs):
+        fixed_kwargs = deepfix_enum_values(kwargs)
+        return method(*args, **fixed_kwargs)
+
+    return wrapper
```

Comment:
> @karisal-anders did you test this feature https://docs.graphene-python.org/projects/django/en/latest/queries/#choices-to-enum-conversion? What about setting the `convert_choices_to_enum = False`?

No, I didn't come across that feature. If it works let's use that instead.

Reply:
It did not work right just how was expected, because it still had issues with the uppercase formatting. While the uppercase formatting is still an issue, it does not make sense to stick with the string inputs, when the enums offers better typing and hinting. I created a new PR of the changes https://github.com/City-of-Helsinki/kukkuu/pull/399. We can take that in action later if needed or wanted.